build.rs

```diff
@@ -1,15 +1,19 @@
 fn main() {
     #[cfg(target_os = "macos")]
     {
-        use cc::Build;
-        use std::path::Path;
+        // check cross-compile target. Only build lladdr.o when actually targeting macOS
+        let target_os = std::env::var("CARGO_CFG_TARGET_OS").unwrap();
+        if target_os == "macos" {
+            use cc::Build;
+            use std::path::Path;
 
-        let path = Path::new("src")
-            .join("target")
-            .join("macos")
-            .join("ffi")
-            .join("lladdr.c");
+            let path = Path::new("src")
+                .join("target")
+                .join("macos")
```

Comment:
Can we have `macos` as a constant as we are using it twice for target?

```rust
const TARGET_MACOS: &str = "macos";
```

And then use:

L6:

```rust
if target_os == TARGET_MACOS
```

L12:

```rust
.join(TARGET_MACOS)
```